In [1]:
import os
import torch
import pandas as pd
import optuna

from definitions import *

In [2]:
model = 'bi-lstm'
# model = 'sent-nn'
optim_path = os.path.join(EXP_DIR_PATH, model, 'optimization')
# study_name = 'bi-lstm_wAtt_sTPE_pNone_df0.2_wf0.2.pkl'
# files_n_params = [sn for sn in optim_path if 'featOptimSentFeat' in sn.split('_')[1]]


In [10]:
# optim_type = 'featOptimFeatModel'
model = '_'.join(model.split('-')) if model == 'sent-nn' else model
# optim_type = 'featOptimFeatModel'
# optim_type = 'featOptimWordLevel'  featCombinationOptimSentFeat
# TODO: store results in pd.DataFrame to easily use with plotly, create awesome charts
for feature_type in ['pos', 'tag', 'dep']:
    # bi-lstm
#     file_name = f'{model}_{optim_type}_{feature_type}_mF1_wAtt_sGrid_pNone_df0.2_wf0.03'
    # sent-nn
    file_name = f'{model}_{optim_type}_{feature_type}_mF1_sGrid_pNone_df0.2_wf0.03'
#     file_name = f'{model}_{optim_type}_mF1_wAtt_sGrid_pNone_df0.2_wf0.03'
    
    study_path = os.path.join(optim_path, f'{file_name}.pkl')
    params_path = os.path.join(optim_path, f'{file_name}_params.pkl')
    study = torch.load(study_path)
    params = torch.load(params_path)
    
    print(feature_type)
    print(study.best_params)
    print(params)

pos
{'stopwords_type': 'wstop', 'from_selection': True, 'pos_feature_type': 'none', 'word_count_feature_type': 'none'}
{'batch_size': 16, 'dropout': 0.28, 'hidden_dim': 128, 'with_sequential_layer': False, 'learning_rate': 0.0008128750160252116, 'optimizer_weigth_decay': 6.849545410012206e-05, 'pos_weight': 32.81159420289855, 'embedding_model_name': 'multi-qa-mpnet-base-dot-v1', 'fnn_hidden_dim': 512, 'fnn_n_hidden_layers': 3, 'fnn_dropout': 0.14, 'pnn_hidden_dim': 32, 'pnn_dropout': 0.23}
tag
{'stopwords_type': 'wstop', 'from_selection': True, 'tag_feature_type': 'sum', 'word_count_feature_type': 'count_words'}
{'batch_size': 16, 'dropout': 0.28, 'hidden_dim': 128, 'with_sequential_layer': False, 'learning_rate': 0.0008128750160252116, 'optimizer_weigth_decay': 6.849545410012206e-05, 'pos_weight': 32.81159420289855, 'embedding_model_name': 'multi-qa-mpnet-base-dot-v1', 'fnn_hidden_dim': 512, 'fnn_n_hidden_layers': 3, 'fnn_dropout': 0.14, 'pnn_hidden_dim': 32, 'pnn_dropout': 0.23}
dep


**bi-lstm feature optim**
- params: {
    'batch_size': 16, 'pooling_strategy': 'last_four', 'dropout': 0.05, 'hidden_dim': 128, 'with_sequential_layer': False, 'learning_rate': 1.0691418898885528e-05, 'optimizer_weigth_decay': 1.0929364050970057e-05, 'pos_weight': 32.81159420289855, 'fnn_hidden_dim': 256, 'fnn_n_hidden_layers': 4, 'fnn_dropout': 0.12, 'pnn_hidden_dim': 8, 'pnn_dropout': 0.24
}

- featOptimSentFeat:
    - pos: {
        'stopwords_type': 'wstop', 'from_selection': True, 'pos_feature_type': 'none', 'word_count_feature_type': 'count_words'
    }
    - tag: {
        'stopwords_type': 'wstop', 'from_selection': True, 'tag_feature_type': 'none', 'word_count_feature_type': 'count_words'
    }
    - dep: {
        'stopwords_type': 'wostop', 'from_selection': True, 'dep_feature_type': 'onehot', 'word_count_feature_type': 'count_words'
    } --> only this one, no need to run another optimizaiton

- featOptimFeatModel:
    - pos: {
        'stopwords_type': 'wstop', 'from_selection': False, 'pos_feature_type': 'sum', 'word_count_feature_type': 'none'
    }
    - tag: {
        'stopwords_type': 'wstop', 'from_selection': False, 'tag_feature_type': 'sum', 'word_count_feature_type': 'none'
    }
    - dep: {
        'stopwords_type': 'wstop', 'from_selection': True, 'dep_feature_type': 'none', 'word_count_feature_type': 'none'
    }

- featOptimWordLevel
    - {
        'stopwords_type': 'wstop', 'word_count_feature_type': 'count_words', 'word_level_feature_type': 'dep'
    }
    
**sent-nn feature optim**
- params: {'batch_size': 16, 'dropout': 0.28, 'hidden_dim': 128, 'with_sequential_layer': False, 'learning_rate': 0.0008128750160252116, 'optimizer_weigth_decay': 6.849545410012206e-05, 'pos_weight': 32.81159420289855, 'embedding_model_name': 'multi-qa-mpnet-base-dot-v1', 'fnn_hidden_dim': 512, 'fnn_n_hidden_layers': 3, 'fnn_dropout': 0.14, 'pnn_hidden_dim': 32, 'pnn_dropout': 0.23}

- featOptimSentFeat:
    - pos: {'stopwords_type': 'wstop', 'from_selection': True, 'pos_feature_type': 'none', 'word_count_feature_type': 'count_words'}
    - tag: {'stopwords_type': 'wstop', 'from_selection': False, 'tag_feature_type': 'sum', 'word_count_feature_type': 'count_words'} --> only this one, no need to run another optimizaiton
    - dep: {'stopwords_type': 'wstop', 'from_selection': False, 'dep_feature_type': 'none', 'word_count_feature_type': 'none'}
    
- featOptimFeatModel:
    - pos: {'stopwords_type': 'wstop', 'from_selection': True, 'pos_feature_type': 'none', 'word_count_feature_type': 'none'}
    - tag: {'stopwords_type': 'wstop', 'from_selection': True, 'tag_feature_type': 'sum', 'word_count_feature_type': 'count_words'}
    - dep: {'stopwords_type': 'wstop', 'from_selection': True, 'dep_feature_type': 'sum', 'word_count_feature_type': 'none'}
    


In [ ]:
# optim_path = 

In [3]:
optim_type = 'featCombinationOptimSentFeat'

# TODO: store results in pd.DataFrame to easily use with plotly, create awesome charts
# file_name = f'bi-lstm_{optim_type}_{feature_type}_mF1_wAtt_sGrid_pNone_df0.2_wf0.03'
file_name = f'{model}_{optim_type}_mF1_wAtt_sGrid_pNone_df0.2_wf0.03'

study_path = os.path.join(optim_path, f'{file_name}.pkl')
params_path = os.path.join(optim_path, f'{file_name}_params.pkl')
study = torch.load(study_path)
params = torch.load(params_path)

print(study.best_params)
print(params)

{'dep_feature': 'word_level'}
{'batch_size': 16, 'pooling_strategy': 'last_four', 'dropout': 0.05, 'hidden_dim': 128, 'with_sequential_layer': False, 'learning_rate': 1.0691418898885528e-05, 'optimizer_weigth_decay': 1.0929364050970057e-05, 'pos_weight': 32.81159420289855, 'fnn_hidden_dim': 256, 'fnn_n_hidden_layers': 4, 'fnn_dropout': 0.12, 'pnn_hidden_dim': 8, 'pnn_dropout': 0.24}
